File: ghis/ghislain-one-app/src/app/business/album.ts
import {Injectable, OnInit} from '@angular/core';
import { Artist } from './artist';
import { Track } from './Track';

declare var $:any;

@Injectable()
export class Album {
    Id:number = 0;
    ArtistId:number = 0;
    Title:string = null;
    Description:string = null;
    Year:number = 0;
    ImageUrl:string = null;
    AmazonUrl:string = null;
    SpotifyUrl:string = null;

    Artist:Artist = new Artist();
    Tracks:Track[] = [];
}

 

 

File: ZoeHome/src/ZoeHome.Domain/Entities/Album.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ZoeHome.Domain.Entities
{
    public class Album
    {
        public Album()
        {
            Artist = new Artist();
            Tracks = new List<Track>();
        }

        public int AlbumId { get; set; }
        public string AmazonUrl { get; set; }
        public virtual Artist Artist { get; set; }
        public int ArtistId { get; set; }
        public string Description { get; set; }
        public string ImageUrl { get; set; }
        public string SpotifyUrl { get; set; }
        public string Title { get; set; }
        public virtual IList<Track> Tracks { get; set; }
        public int Year { get; set; }
    }

    public class ArtistWithAlbumCount : Artist
    {
        public int AlbumCount { get; set; }
    }
}